### PR TITLE
#CP-21 feat:  Logout from account

### PR DESCRIPTION
**Pull request by samyshyaka**
#### What does this PR do?
Gives an authenticated user the ability to logout, so that none else can access their account as well.

#### Description of the tasks to be completed
- [x] Authenticated user should be able to logout.
- [x] Authenticated user should be able to access protected resourses/URLs.
- [x] Appropriate error messages are displayed on unsuccessful attempts.

#### How should this be manually tested?
**Configure your environment based on the `.env.sample.md` then follow the following:**
```js
git checkout ft-logout-cp-21
git fetch
npm install
// if you don't have postgres installed, refer to the README for instructions
npm start
```
> **Testing the login route**
Use the below endpoints
```
POST   /api/v1/users    //Creating a user with {firstName, lastName, email & password}
POST   /api/v1/login    //Sign in with {email & password}
GET   /api/v1/logout    //Logout
```
#### Pre-requisites:

- Clone the project
- Make sure that Postgres is up and running

#### What are the relevant trello stories?
[#CP-21](https://trello.com/c/bhS4kERe/21-logout-from-account)

### CODE REVIEW

File: src/controllers/logout.controller.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const logout = async (req, res) => {
+	const { cookies } = req;
+	const accessToken = req.headers.authorization;
+	if (!cookies.jwt && !accessToken) {
+		return handleResponse(res, 200, { message: res.__('logout successful') });
+	}
+
+	/* c8 ignore next 7 */
+	if (cookies.jwt) {
+		await DataSource.query('TRUNCATE TABLE "RefreshToken"');
```
Comment (feyton) [inline src/controllers/logout.controller.js]: There is no need to drop the while table while deleting one refresh token. 
You have to check for just a specifically a token and remove that.

**Review by feyton — CHANGES_REQUESTED** [review]
@samuel-shyaka-dus 
This is great work. I left a review that needs special attention that affects the implementation. 
In the meantime that is great work.

**Review by feyton — DISMISSED** [review]

**Review by feyton — DISMISSED** [review]

**Review by Yvad60 — DISMISSED** [review]

**Review by Abdulkeza — DISMISSED** [review]
Great job sam, I have seen that all the issues you had 3 hours ago were solved, now it is working as expected

**Review by Yvad60 — APPROVED** [review]

**Review by feyton — APPROVED** [review]
@samuel-shyaka-dus 
This is really a great work. The feature is well implemented. as we will integrate **Redis** into the project in the coming days. We will add a **token blacklist** so that users will be able to terminate their access_token before time which will be a great user experience. But as we are giving the token short life span. This is really great.

These changes can be merged.

**Review by Jkadhuwa — APPROVED** [review]

**Pull request merged**